caas/kubernetes/provider/specs/admissionregistration.go

```diff
@@ -57,6 +57,71 @@ func (wh K8sMutatingWebhookSpec) MarshalJSON() ([]byte, error) {
 	}
 }
 
+// UpgradeK8sMutatingWebhookSpecV1Beta1 converts a v1beta1 MutatingWebhook to v1.
+func UpgradeK8sMutatingWebhookSpecV1Beta1(spec admissionregistrationv1beta1.MutatingWebhook) admissionregistrationv1.MutatingWebhook {
+	hook := admissionregistrationv1.MutatingWebhook{
+		Name:                    spec.Name,
+		NamespaceSelector:       spec.NamespaceSelector,
+		ObjectSelector:          spec.ObjectSelector,
+		TimeoutSeconds:          spec.TimeoutSeconds,
+		AdmissionReviewVersions: spec.AdmissionReviewVersions,
+		ClientConfig: admissionregistrationv1.WebhookClientConfig{
+			URL:      spec.ClientConfig.URL,
+			CABundle: spec.ClientConfig.CABundle,
+		},
+	}
+	if len(hook.AdmissionReviewVersions) == 0 {
+		hook.AdmissionReviewVersions = []string{"v1beta1"}
```

Comment:
Just checking - we want this "v1beta1" in a v1 CRD?

Reply:
Yes, this is the outgoing hook format (i.e. old apis that *receive* hooks want it in the old format)